model-optimizer/extensions/front/onnx/ONNXResize10ToInterpolate.py

```diff
@@ -0,0 +1,139 @@
+                                           'nearest_mode': 'simple',
+                                           'pads_begin': int64_array([0]),
+                                           'pads_end': int64_array([0]),
+                                           'antialias': 0,
+                                           'shape_calculation_mode': 'scales',
+                                           'in_ports_count': 4}).create_node()
+
+    range_node.out_port(0).connect(interpolate_node.in_port(3))
+    shape_of = Shape(graph, {'name': resize_name + '/ShapeOf'}).create_node()
+
+    # When we calculate 'sizes' input as floor(input_shape * scales), we can get incorrect 'sizes' if, e.g.,
+    # scales = [1.0, 1.0, 1.33333, 2.0], input_shape = [1, 3, 30, 200], because
+    # input_shape * scales = [1, 3, 39.9999, 400], and floor(input_shape * scales)[2] == 39, not 40.
+    # Maybe we need to calculate 'sizes' input as floor(input_shape * scales + eps), where eps is some small
+    # floating point number, e.g. 1.0e-5. But, in this case, if scales = [1.0, 1.0, 1.333333, 2.0],
+    # input_shape = [1, 3, 30, 200], floor(input_shape * scales + eps) = 39, not 40, because
+    # input_shape[2] * scales[2] + 1.0e-5 =  39.99991.
+    # Hence, we need to calculate 'sizes' as floor(input_shape * (scales + eps)).
+    add_node = create_op_with_const_inputs(graph, Add,
+                                           {1: float_array([1.0e-5])},
+                                           {'name': resize_name + '/Add'})
+
+    input_data_type = data_type_str_to_np(graph.graph['cmd_params'].data_type)
+
+    cast_shape_to_float = Cast(graph, {'dst_type': input_data_type}).create_node()
+    mul_node = Mul(graph, {'name': resize_name + '/Mul'}).create_node()
+    floor_node = Floor(graph, {'name': resize_name + '/Floor'}).create_node()
+    cast_mul_result_to_int = Cast(graph, {'dst_type': np.int64}).create_node()
+
+    shape_of.out_port(0).connect(cast_shape_to_float.in_port(0))
+    cast_shape_to_float.out_port(0).connect(mul_node.in_port(0))
+    add_node.out_port(0).connect(mul_node.in_port(1))
+    mul_node.out_port(0).connect(floor_node.in_port(0))
+    floor_node.out_port(0).connect(cast_mul_result_to_int.in_port(0))
+    cast_mul_result_to_int.out_port(0).connect(sizes_ss.in_port(0))
+    sizes_ss.out_port(0).connect(interpolate_node.in_port(1))
+
+    scales_ss.out_port(0).connect(interpolate_node.in_port(2))
+
+    connection_of_resize_input = resize.in_port(0).get_connection()
+    connection_of_resize_input.set_destination(interpolate_node.in_port(0))
+
+    connection_of_scales = resize.in_port(1).get_connection()
+    connection_of_scales.set_destination(scales_ss.in_port(0))
+
+    connection_of_resize_input.get_source().connect(shape_of.in_port(0))
+    connection_of_resize_input.get_source().connect(rank_node.in_port(0))
+    connection_of_scales.get_source().connect(add_node.in_port(0))
```

Comment:
Since this is a front phase transformation we can get rid of ports manipulations in many cases. **If we know that the output port of operations node "x" and "y" are 0** and we want to connect it as inputs to operation "Z" then we can write the code like this:
 ```python 
Z(graph, {attrs}).create_node([x, y])
``` 

instead of
```python
z = Z(graph, {attrs})
x.out_port(0).connect(z.in_port(0))
y.out_port(0).connect(z.in_port(1))
```

Reply:
Done.